build.gradle

```diff
@@ -11,11 +11,55 @@ repositories {
 }
 
 dependencies {
-    implementation "org.jetbrains.kotlin:kotlin-stdlib"
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.1'
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.1'
+    implementation("org.jetbrains.kotlin:kotlin-stdlib")
+    implementation("org.apache.maven:maven-artifact")
+    implementation("io.github.microutils:kotlin-logging-jvm:2.1.21")
+
+    testImplementation("org.junit.jupiter:junit-jupiter-api")
+    testImplementation("org.assertj:assertj-core")
+
+    testImplementation("org.junit.jupiter:junit-jupiter-params")
+
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine")
+
+    constraints {
+        api("org.apache.maven:maven-artifact:3.5.0")
+        api("org.junit.jupiter:junit-jupiter-api:$junitJupiterVersion")
+        api("org.junit.jupiter:junit-jupiter-engine:$junitJupiterVersion")
+        api("org.assertj:assertj-core:3.22.0")
+    }
+}
+
+tasks.named("compileTestKotlin").configure {
+    kotlinOptions {
+        freeCompilerArgs += "-opt-in=kotlin.io.path.ExperimentalPathApi"
+    }
 }
 
 test {
     useJUnitPlatform()
+    inputs.dir(layout.projectDirectory.dir("./src/test/fixtures"))
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
```

Comment:
Since gradle still supports Java 8 the plugin should be compatible as well:
https://docs.gradle.org/current/userguide/compatibility.html#:~:text=A%20Java%20version%20between%208,used%20for%20compile%20or%20test.